### ISSUE TITLE
Add the ability to list files to be tested

### ISSUE THREAD

**Issue by plroebuck** [mochajs/mocha#3581]
### Description

Output files to be tested (in load order) with the [debug](https://www.npmjs.com/package/debug) package using the namespace `'mocha:files'` in ["bin/_mocha"](https://github.com/plroebuck/mocha/blob/3fbbb0267e0420f353dfa916069b0f19e8dc4f09/bin/_mocha#L575) file.

```js
var debug = require('debug')('mocha:files');    // immediately before L16
...
debug('files:', files);    // L575
```

### Additional Information

Test as follows:
1. Using default "test" directory
```bash
$ DEBUG='mocha:files' mocha --recursive
```
2. Using extended glob (results should be same as case 1)
```bash
$ DEBUG='mocha:files' mocha 'test/**/*.js'
```
3. Using default "test" directory, excluding files in subdir "setup"
```bash
$ DEBUG='mocha:files' mocha --exclude 'test/setup/**' --recursive
```
4. Using extended glob, excluding files in subdir "setup" (results should be same as case 3)
```bash
$ DEBUG='mocha:files' mocha --exclude 'test/setup/**' 'test/**/*.js'
```
5. Using default "test" directory, with prepended file "test/setup/init.js"
```bash
$ DEBUG='mocha:files' mocha --file test/setup/init.js --exclude 'test/setup/**' --recursive
```
6. Using extended glob, with prepended file "test/setup/init.js" (results should be same as case 5)
```bash
$ DEBUG='mocha:files' mocha --file test/setup/init.js --exclude 'test/setup/**' 'test/**/*.js'
```

**Comment by kennycastro007** [comment]
Anybody mind if I take this one?

**Comment by plroebuck** [comment]
Exists in Mocha-6.0 (though hate the namespace) using:
```sh
$ DEBUG='mocha:cli:run:helpers' mocha --recursive
```

**Comment by kennycastro007** [comment]
So should someone close this issue? Since it's already fixed now?

**Comment by plroebuck** [comment]
I'll leave it open for now and see if I can use a simpler namespace...

**Comment by boneskull** [comment]
@plroebuck 	I'm confused; is this intended to be user-facing?  If so, `DEBUG` is sort of a wonky way to do it.  It *would* be cool to see:

1. A list of all test files
1. A tree of all test files and user code (stuff *not* in a `node_modules`)
1. A `--dry-run` which would dump all suites but not run any tests

**Comment by boneskull** [comment]
Given the cross-cutting concerns that such an implementation would need to touch, I don't think this is a good candidate for a first issue

**Comment by johnwhelchel** [comment]
This was implemented and closed quite quickly it seems. 

Can we re-open this PR?

It seems like something a lot of people would like!

https://github.com/mochajs/mocha/pull/1070

**Comment by sh3raawii** [comment]
You are welcome
https://github.com/sh3raawii/mocha-decaf

**Comment by JoshuaKGoldberg** [comment]
Fixed by #4640. 🚀
